File: src/main/java/com/telegram/codex/bootstrap/TelegramCodexApplication.java
package com.telegram.codex.bootstrap;

import com.telegram.codex.shared.config.AppProperties;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.context.properties.EnableConfigurationProperties;

import java.nio.file.Files;
import java.nio.file.Path;

@SpringBootApplication(scanBasePackages = "com.telegram.codex")
@EnableConfigurationProperties(AppProperties.class)
public class TelegramCodexApplication {

    public static void main(String[] args) {
        SpringApplication application = new SpringApplication(TelegramCodexApplication.class);
        createDataDirectory();
        application.run(args);
    }

    private static void createDataDirectory() {
        String configuredPath = System.getenv().getOrDefault("SQLITE_DB_PATH", "./data/app.db");
        Path sqlitePath = Path.of(configuredPath).toAbsolutePath().normalize();
        try {
            Files.createDirectories(sqlitePath.getParent());
        } catch (java.io.IOException error) {
            throw new IllegalStateException("Failed to create SQLite directory for " + sqlitePath, error);
        }
    }
}

File: src/main/kotlin/com/telegram/codex/bootstrap/TelegramCodexApplication.kt
package com.telegram.codex.bootstrap

import com.telegram.codex.shared.AppProperties
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.AutoConfigurationPackage
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.boot.persistence.autoconfigure.EntityScan
import java.io.IOException
import java.nio.file.Files
import java.nio.file.Path

@SpringBootApplication(scanBasePackages = ["com.telegram.codex"])
@AutoConfigurationPackage(basePackages = ["com.telegram.codex"])
@EntityScan(basePackages = ["com.telegram.codex"])
@EnableConfigurationProperties(AppProperties::class)
class TelegramCodexApplication

fun main(args: Array<String>) {
    val application = SpringApplication(TelegramCodexApplication::class.java)
    createDataDirectory()
    application.run(*args)
}

private fun createDataDirectory() {
    val configuredPath = System.getenv().getOrDefault("SQLITE_DB_PATH", "./data/app.db")
    val sqlitePath = Path.of(configuredPath).toAbsolutePath().normalize()
    try {
        Files.createDirectories(sqlitePath.parent)
    } catch (error: IOException) {
        throw IllegalStateException("Failed to create SQLite directory for $sqlitePath", error)
    }
}
